mono/mini/debug-mini.c

```diff
@@ -616,13 +621,18 @@ mono_debug_add_aot_method (MonoDomain *domain, MonoMethod *method, guint8 *code_
 	if (debug_info_len == 0)
 		return;
 
-	jit = deserialize_debug_info (method, code_start, debug_info, debug_info_len);
-
-	mono_debug_add_method (method, jit, domain);
-
-	mono_debug_add_vg_method (method, jit);
-
-	mono_debug_free_method_jit_info (jit);
+	/* Cache the data here, and decode it when its needed in mini_debug_find_jit_debug_info () */
+	AotDebugInfo *info = mono_domain_alloc0 (domain, sizeof (AotDebugInfo));
```

Comment:
I think this line -- needs a cast in C++.

Reply:
This line does not need a cast. It is the other one. We could make g_hash_table_lookup "auto cast" though.